lambda/openhab/index.js

```diff
@@ -0,0 +1,250 @@
+      //  - org.eclipse.smarthome.i18n (OH 2.5) [API Version == 3]
+      //  - org.eclipse.smarthome.core.i18nprovider (OH 2.0 -> 2.4) [API Version <= 2]
+      const serviceId = apiVersion === 3 ? 'org.eclipse.smarthome.i18n' : 'org.eclipse.smarthome.core.i18nprovider';
+      const { language, measurementSystem, region } = await this.getServiceConfig(serviceId);
+      return {
+        regional: { language, measurementSystem, region },
+        runtime: { uuid, version: '2' }
+      };
+    }
+  }
+
+  /**
+   * Sends a command to an item and cache its value
+   * @param  {String}  itemName
+   * @param  {String}  command
+   * @return {Promise}
+   */
+  sendCommand(itemName, command) {
+    // Cache posted command
+    this._cache.postedCommands[itemName] = command;
+    return this.postItemCommand(itemName, command);
+  }
+
+  /**
+   * Returns a single item
+   * @param  {String}  itemName
+   * @return {Promise}
+   */
+  getItem(itemName) {
+    const options = {
+      method: 'GET',
+      uri: `/rest/items/${itemName}`,
+      json: true
+    };
+    return this._request(options);
+  }
+
+  /**
+   * Returns all items with alexa, channel and synonyms metadata
+   * @return {Promise}
+   */
+  getAllItems() {
+    const options = {
+      method: 'GET',
+      uri: '/rest/items',
+      qs: {
+        fields: 'editable,groupNames,groupType,name,label,metadata,stateDescription,tags,type',
+        metadata: 'alexa,autoupdate,channel,synonyms'
+      },
+      json: true
+    };
+    return this._request(options);
+  }
+
+  /**
+   * Returns root resource
+   * @return {Promise}
+   */
+  getRootResource() {
+    const options = {
+      method: 'GET',
+      uri: '/rest',
+      json: true
+    };
+    return this._request(options);
+  }
+
+  /**
+   * Returns a service config
+   * @param  {String}  serviceId
+   * @return {Promise}
+   */
+  getServiceConfig(serviceId) {
+    const options = {
+      method: 'GET',
+      uri: `/rest/services/${serviceId}/config`,
+      json: true
+    };
+    return this._request(options);
+  }
+
+  /**
+   * Returns uuid
+   * @return {Promise}
+   */
+  getUUID() {
+    const options = {
+      method: 'GET',
+      uri: '/rest/uuid'
+    };
+    return this._request(options);
+  }
+
+  /**
+   * Sends a command to an item
+   * @param  {String}  itemName
+   * @param  {String}  value
+   * @return {Promise}
+   */
+  postItemCommand(itemName, value) {
+    const options = {
+      method: 'POST',
+      uri: `/rest/items/${itemName}`,
+      headers: {
+        'Content-Type': 'text/plain'
+      },
+      body: value.toString()
+    };
+    return this._request(options);
+  }
+
+  /**
+   * Returns item state formatted based on its state description pattern and type
+   * @param  {Object} item
+   * @return {String}
+   */
+  static formatItemState(item) {
+    const format = item.stateDescription && item.stateDescription.pattern.match(/%(?:[.0]\d+)?[dfs]/);
```

Comment:
```suggestion
    const format = item.stateDescription &&  item.stateDescription.pattern && item.stateDescription.pattern.match(/%(?:[.0]\d+)?[dfs]/);
```
Just testing now,  I was getting an error that item.stateDescription.pattern was undefined here when calling `match` , this seemed to fix it.

Reply:
Thanks I will include the fix. There are other instances where I made the assumption that the `pattern` is always present if `stateDescription` is available.

I made the same assumption with `options` but from my testing, it seems that this property is always present.